src/libraries/System.Formats.Tar/src/System/Formats/Tar/PaxTarEntry.cs

```diff
@@ -90,6 +90,21 @@ public PaxTarEntry(TarEntryType entryType, string entryName, IEnumerable<KeyValu
             _header.ReplaceNormalAttributesWithExtended(extendedAttributes);
         }
 
+        /// <summary>
+        /// Initializes a new <see cref="PaxTarEntry"/> instance by converting the specified <paramref name="other"/> entry into the PAX format.
+        /// </summary>
+        public PaxTarEntry(TarEntry other)
+            : this(other._header, other._readerOfOrigin!)
+        {
+            if (_header._typeFlag == TarEntryType.V7RegularFile)
+            {
+                _header._typeFlag = TarEntryType.RegularFile;
+            }
+            TarHelpers.VerifyEntryTypeIsSupported(_header._typeFlag, TarEntryFormat.Pax, forWriting: false);
+
+            _header._format = TarEntryFormat.Pax;
+        }
```

Comment:
If `other` is a GnuTarEntry that has an atime or ctime value, shouldn't those end up somewhere in the new entry's extended attributes?

Similarly, what about the PosixTarEntry fields?  Nothing seems to be copying over the UserName/GroupName values.

Reply:
In line 97, `this(other.header, ...)` calls `base(other.header, ...)` which then saves that header in this entry. The header struct contains _all_ the info.

If the user writes that entry to a `TarWriter`, the entry will be written with its fields filled out as needed by the format of that entry.

Reply:
Ok yeah, I'll add the logic, it wasn't happening anyway before writing. The extended attributes were only being collected when reading for an existing archive. Need to do the same when they are created manually, or are being converted.

Reply:
From our chat conversation, I need to check that the extended attributes are collected properly in PAX, and I also need to make sure that converting from one format to another drops the data that is not needed in that format.

Reply:
Please see the newest commits. I changed the code so that only the supported fields are transferred when converting between formats.